### PR TITLE
docs: extend testing documentation

### PR DESCRIPTION
**Pull request by kyubisation**

### CODE REVIEW

**Comment by kyubisation** [comment]
Could anyone using IntelliJ add the section about debugging/testing in it?

File: DEVELOPER.md [hunk 1]
```diff
@@ -29,7 +29,43 @@ Run `yarn lint --fix` to fix any automatically fixable lint issues and report th
...
+It is possible to debug tests and/or run them in isolation with Visual Studio Code.
+The following code snippet can be placed in `.vscode/launch.json`.
+Replace `test:csr` with either `test:ssr:hydrated` or `test:ssr:non-hydrated` to test SSR.
+
+```json
+  ...
+  {
+    "name": "Test",
+    "request": "launch",
+    "runtimeArgs": ["test:csr", "${relativeFile}", "--watch"],
```
Comment (TomMenga) [inline DEVELOPER.md:48]:
```suggestion
    "runtimeArgs": ["test:csr", "${relativeFile}", "--watch", "--debug"],
```

I am not 100% sure we should add the `--debug` flag in the documentation but, if we don't, you cannot use breakpoints and won't have the full detailed summary report.
I'm leaning toward adding it though

**Review by jeripeierSBB — APPROVED** [review]
LGTM

File: DEVELOPER.md [hunk 2]
```diff
@@ -29,7 +29,43 @@ Run `yarn lint --fix` to fix any automatically fixable lint issues and report th
...
+    "runtimeExecutable": "yarn",
+    "skipFiles": ["<node_internals>/**"],
+    "type": "node",
+    "console": "integratedTerminal"
+  },
+  ...
```
Comment (kyubisation) [inline DEVELOPER.md:54]:
```suggestion
  {
    "name": "Test Debug",
    "request": "launch",
    "runtimeArgs": ["test:csr", "${relativeFile}", "--watch", "--debug"],
    "runtimeExecutable": "yarn",
    "skipFiles": ["<node_internals>/**"],
    "type": "node",
    "console": "integratedTerminal"
  },
  ...
```

File: DEVELOPER.md (hunk 1 above)
Comment (kyubisation) [reply DEVELOPER.md:48]: We could add a second section. What do you think?
https://github.com/lyne-design-system/lyne-components/pull/2445/files#r1507216791

File: DEVELOPER.md [hunk 3]
```diff
@@ -29,7 +29,43 @@ Run `yarn lint --fix` to fix any automatically fixable lint issues and report th
...
+To run unit tests, run `yarn test`. This will run the full test suite with client side rendering
+and server side rendering with and without hydration.
+During development it is preferable to run `yarn test:csr`, which will just run the client side
+rendering.
+
+### Debugging with Visual Studio Code
+
+It is possible to debug tests and/or run them in isolation with Visual Studio Code.
+The following code snippet can be placed in `.vscode/launch.json`.
+Replace `test:csr` with either `test:ssr:hydrated` or `test:ssr:non-hydrated` to test SSR.
```
Comment (TomMenga) [inline DEVELOPER.md:41]:
```suggestion
Replace `test:csr` with either `test:ssr:hydrated` or `test:ssr:non-hydrated` to test SSR.
Add the `--debug` param to enable breakpoint debugging and the detailed test report.
```

Or we could just add a phrase here

Comment (kyubisation) [reply DEVELOPER.md:41]: Done 👍

**Comment by kyubisation** [comment]
By the way, I realized `--filter` does nothing. You can just provide a single path or a list of paths and that will act as a filter.

**Pull request merged**